cvat/apps/engine/cache.py

```diff
@@ -54,74 +62,248 @@
     ZipChunkWriter,
     ZipCompressedChunkWriter,
 )
-from cvat.apps.engine.utils import load_image, md5_hash
+from cvat.apps.engine.rq_job_handler import RQJobMetaField
+from cvat.apps.engine.utils import get_rq_lock_for_job, load_image, md5_hash
 from utils.dataset_manifest import ImageManifestManager
 
 slogger = ServerLogManager(__name__)
 
 
 DataWithMime = Tuple[io.BytesIO, str]
-_CacheItem = Tuple[io.BytesIO, str, int]
+_CacheItem = Tuple[io.BytesIO, str, int, Union[datetime, None]]
+
+
+class CvatCacheTimestampMismatchError(Exception):
+    pass
+
+
+def enqueue_create_chunk_job(
```

Comment:
Please reflect in the function interface that it also waits for the result, it's not obvious.